centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -286,18 +247,33 @@ object Operations {
     // For JES it should be fine but locally it can be quite fast
     def pollDelay() = blocking { Thread.sleep(5000) }
 
+    // Special case for sub workflow testing
+    def findJobIdInSubWorkflow(subWorkflowId: String): Option[String] = {
+      for {
+        metadata <- CentaurCromwellClient.metadata(WorkflowId.fromString(subWorkflowId)).toOption
+        jobId <- metadata.value.get("calls.inner_abort.aborted.jobId")
+      } yield jobId.asInstanceOf[JsString].value
```

Comment:
ToL  wanted to point you at shapeless support for casting: https://github.com/milessabin/shapeless/wiki/Feature-overview:-shapeless-2.0.0#type-safe-cast

AFAIK it just wraps the cast and returns an `Option` so it's marginally safer.

Reply:
Oh yep that's better - because this is pre-existing I'll leave it like this for now but will use my bug rotation time to make a second pass later